docs/test-prep.md

```diff
@@ -0,0 +1,406 @@
+# Preparing a test for MFTF
+
+This tutorial will guide you through the process of converting a raw functional test into a properly abstracted test file, ready for publishing.
+
+## The abstraction process
+
+When first writing a test for a new piece of code such as a custom extension, it is likely that values are hardcoded for the specific testing environment while in development. To make the test more generic and easier for others to update and use, we need to abstract the test.
+The general process:
+
+1. Create the raw test.
+1. Update all the selectors from hardcoded values to appropriate selectors.
+1. Update userInput from hardcoded values to data entities.
+1. Remove simple actions and replace with Action Groups.
+
+## The manual test
+
+Manual tests are just that: A series of manual steps to be run.
+
+```xml
+<!-- Navigate to Catalog -> Products page (or just open by link) -->
+<!-- Fill field "Name" with "Simple Product %unique_value%" -->
+<!-- Fill field "SKU" with "simple_product_%unique_value%" -->
+<!-- Fill field "Price" with "500.00" -->
+<!-- Fill field "Quantity" with "100" -->
+<!-- Fill field "Weight" with "100" -->
+<!-- Click "Save" button -->
+<!-- See success save message "You saved the product." -->
+<!-- Navigate to Catalog -> Products page (or just open by link) -->
+<!-- See created product is in grid -->
+<!-- See "Name" in grid is valid -->
+<!-- See "SKU" in grid is valid -->
+<!-- See "Price" in grid is valid -->
+<!-- See "Quantity" in grid is valid -->
+<!-- Open Storefront Product Page and verify "Name", "SKU", "Price" -->
+```
+
+## The raw test
+
+This tests works just fine. But it will only work if everything referenced in the test stays exactly the same. This neither reusable nor extensible.
+Hardcoded selectors make it impossible to reuse sections in other actiongroups and tasks. They can also be brittle. If Magento happens to change a `class` or `id` on an element, the test will fail.
+
+Some data, like the SKU in this example, must be unique for every test run. Hardcoded values will fail here. Data Entities allow for `suffix` and `prefix` for ensuring unique data values.
```

Comment:
Add a link to the topic about data entities.

Reply:
Done